### DISCUSSION TITLE
请问如何引入echarts，实现图表功能

### DISCUSSION THREAD

**Discussion by bule-kimi** [vue-mini/vue-mini#118] [Q&A] [answered]
项目中需要用到echarts图表功能，目前引入之后，页面展示空白。是否支持引入echarts图表

**Comment by yangmingshan** [comment]
echarts 支持小程序吗？有提供小程序版吗？

**Reply by bule-kimi** [reply]
嗯呢 ，echarts官网有提供小程序版本的，具体可以参考下这个地址，https://echarts.apache.org/handbook/zh/how-to/cross-platform/wechat-app

**Reply by yangmingshan** [reply]
需要你提供一个 repo

**Reply by yangmingshan** [reply, marked as answer]
试了一下，并没有发现什么问题，可以正常使用。
https://github.com/yangmingshan/vue-mini-echarts

**Comment by bule-kimi** [comment]
我这边似乎是计算的问题，页面引入四个子组件，然后将图表渲染封装成另外的组件wx-echarts，其中这四个组件共用wx-echarts组件，然后ec的值，用计算方式，发现当页面下的子组件请求接口回来后，ec没有再次更新值

**Reply by bule-kimi** [reply]
import * as echarts from '../ec-canvas/echarts.min';
import { defineComponent, computed } from '@vue-mini/core';

defineComponent({
  properties: {
    height: {
      type: String,
    },
    option: {
      type: Object,
    },
    echartsId: {
      type: String
    }
  },
  setup(props: { echartsId: string; option: object }, context) {
    let chart = null;
    const initChart = (canvas, width: number, height: number, dpr: number) => {
      chart = echarts.init(canvas, null, {
        width: width,
        height: height,
        devicePixelRatio: dpr, //  像素比
      });
      canvas.setChart(chart);

      const _option = props.option;
      chart.setOption(_option);
      return chart;
    };

    const chartId = computed(() => `chart_${props.echartsId}`);
    const canvasId = computed(() => `chart_${props.echartsId}`);

    const ec = computed(() => {
      return {
        lazyload: true,
        onInit: initChart,
      };
    });

    return { ec, chartId, canvasId, initChart };
  },
});

**Reply by yangmingshan** [reply]
ec 主体是个 function 为什么要用 computed 呢？

**Reply by yangmingshan** [reply]
看起来是你使用的问题

**Reply by bule-kimi** [reply]
但是不用computed，也无法及时获取父组件传递的数据，当父组件的值更新后，ec没有同步更新

**Reply by yangmingshan** [reply]
闭包就可以吧

**Reply by bule-kimi** [reply]
const ec = ref({
    onInit: initChart
  });如果直接这个样子的话，拿到的只是最开始的值

**Reply by yangmingshan** [reply]
请给一个最小的完整的可以运行的复现（repo），说明期望表现是什么，实际表现是什么。我才能帮到你。

**Reply by bule-kimi** [reply]
谢谢，这边好像找到原因了，同样的功能，模拟接口返回数据，数据有关系，如果是实际情况接口的话，接口数据量过大，会导致上面的问题
